Mobile Release v1.47.0

antonis opened the pull request:
## Description
Release 1.47.0 of the react-native-editor and Gutenberg-Mobile.

For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3155

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/native-mobile.md -->

Commit:
[RNMobile] Ignore column width attribute when empty (#29015)

* Ignore empty width attribute in columns

The editor was crashing when attempting to render a column block which contained an empty width attribute (i.e `<!-- wp:column {"width":""} -->`.
The exception occurred when accessing a property of an undefined object, so adding a default object (`|| {}`) fixes this by allowing the property to be accessed.

* Update react-native-editor changelog

* Add missing dependency to useSelect in columns block

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

antonis committed:
Release script: Update react-native-editor version to 1.47.0

antonis committed:
Release script: Update with changes from 'npm run core preios'

dcalhoun commented on packages/react-native-bridge/package.json:
```diff
@@ -1,31 +1,31 @@
 {
-  "name": "@wordpress/react-native-bridge",
```
Based on https://github.com/WordPress/gutenberg/pull/28634#discussion_r568479085, it looks like it may be a good idea for us to manually change the `package.json` files back to tabs in our editor.

dcalhoun replied:
I think I am mistaken on this. This appears to be the inverse of what I said. I think these changes should be left as is.

dcalhoun committed:
Update editor changelog

dcalhoun reviewed (commented):
This is looking good to me so far.

antonis reviewed (commented):
LGTM 🎉

antonis committed:
Adds null check before function call

Commit:
Mobile - RichText - Restore onSelectionChange when its focused (#29074)

fluiddot committed:
[RNMobile] Merge 1.46.1 beta fix release to 1.47.0 (#29044)

* Release script: Update react-native-editor version to 1.46.1

* Release script: Update with changes from 'npm run core preios'

* Update react-native-editor CHANGELOG

* [RNMobile] Add capability to bridge to show/hide audio block (#28952)

* [RNMobile] Ignore column width attribute when empty (#29015)

* Update react-native-editor CHANGELOG

* Fixes minor changelog alignement issue

* Add Stories bridge methods for iOS (#29083)

Add missing bridge methods for the Stories block. The absence of these
threw an error, even though Stories block isn't fully supported for iOS
just yet.

Co-authored-by: Ceyhun Ozugur <ceyhunozugur@gmail.com>
Co-authored-by: Paul Von Schrottky <paul.von.schrottky@automattic.com>
Co-authored-by: Antonis Lilis <antonis.lilis@automattic.com>
Co-authored-by: David Calhoun <dpcalhoun@gmail.com>

antonis commented:
At this stage this LGTM 🎉 
Please note that this should not be merged yet. To proceed with this we should **also wait** [`1.46.2`](https://github.com/WordPress/gutenberg/pull/29107) to merge.

dcalhoun reviewed (approved):
LGTM. However, before we merge, we need to...

1. Determine our strategy to fix/validate the current CI checks on this PR stuck in a pending state.
2. Complete the remaining release checklist tasks to merge the `1.47.0` release into WPiOS and WPAndroid first.

After this, we can resolve conflicts here and merge this PR.

antonis committed:
Merge branch 'master' into rnmobile/release_1.47.0

dcalhoun reviewed (approved):
LGTM 🎉
